ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -211,25 +228,50 @@ final class ApiSubmissionService private (
           contractStore.lookupContractKey(commands.submitter, _),
           commands.commands
         )
-      submissionResult <- handleResult(res)
+      transactionInfo <- res.fold(error => {
+        Metrics.failedInterpretationsMeter.mark()
+        Future.failed(grpcError(toStatus(error)))
+      }, Future.successful)
+      partyAllocationResults <- if (configuration.implicitPartyAllocation) {
+        allocateParties(transactionInfo)
+      } else {
+        Future.successful(Seq.empty)
+      }
+      submissionResult <- partyAllocationResults.find(_ != SubmissionResult.Acknowledged) match {
```

Comment:
With the usual recommendation of moving nested logic in a for-comprehension to its own method, I would favor something like
```
if (!partyAllocationResults.contains(SubmissionResult.Acknowledged))
  submitTransaction(transactionInfo)
else
  Future.successful(result)
```

Reply:
My approach of course has the downside of necessarily traversing the whole collection, so use your judgement.

Reply:
Problem is that `result` comes from the return value of `find`, so this doesn't help.